src/types.ts

```diff
@@ -0,0 +1,28 @@
+export type AuthorizationUrlParams = {
+  state?: string
+  scope?: string | string[]
+  nonce?: string | null
```

Comment:
Just did my research and turns out optional params will still throw error if `null` is provided. Leaving this intentional comment here as a learning point for myself!

Reply:
yup haha because sometime there's occasion to leave the parameter empty or intentionally leave it as `null | undefined`. i think in this case it's for users who want to explicitly leave the nonce as null

Reply:
? refers to undefined. null !== undefined